### PR TITLE
added test not allowing two sitelinks to the same site T52362

### PR DESCRIPTION
**Pull request by WMDE-Fisch**

### CODE REVIEW

**Comment by zeljkofilipin** [comment]
:+1:

File: tests/browser/features/sitelinks.feature [hunk 1]
```diff
@@ -219,7 +219,19 @@ Feature: Edit sitelinks
...
+      And I type en into the 1. siteid input field
       And I type xyz_nonexistentarticle_xyz into the page input field
       And I click the sitelink save button
     Then An error message should be displayed for sitelink group wikipedia
+
+  @save_sitelink
+  Scenario: Add new sitelink to already referenced site
+    Given The following sitelinks do not exist:
+      | enwiki | Asia |
+    When I add the following sitelinks:
```
Comment (JeroenDeDauw) [inline tests/browser/features/sitelinks.feature]: why is there `enwiki` above and `en` below?

Comment (WMDE-Fisch) [inline tests/browser/features/sitelinks.feature]: The first one calls the API, where the whole key is checked. The second one uses the frontend where the usage of the complete key is kind of undesired, and the "en" will be auto completed internally. 

That should be refactored in the browsertests though. I will put this in an other issue.

File: tests/browser/features/step_definitions/sitelinks_steps.rb [hunk 2]
```diff
@@ -40,10 +40,10 @@
   end
 end
 
-When /^I type (.+) into the siteid input field$/ do |value|
+When /^I type (.+) into the (\d+). siteid input field$/ do |value, num|
   on(ItemPage) do |page|
-    page.site_id_input_field_wd.clear
-    page.site_id_input_field_wd.send_keys value
+    page.site_id_input_field_wd(num).clear
+    page.site_id_input_field_wd(num).send_keys value
```
Comment (zeljkofilipin) [inline tests/browser/features/step_definitions/sitelinks_steps.rb]: Is this clear/send_keys dance necessary? Just setting the element does not do the job? Something like:

    page.site_id_input_field_wd(num).set value

Comment (WMDE-Fisch) [inline tests/browser/features/step_definitions/sitelinks_steps.rb]: The WebDriver::Element we use here does not support the set method. Furthermore I think that send_keys is needed in some cases for the frontend JS to detect changes in the input field.

Comment (tobijat) [inline tests/browser/features/step_definitions/sitelinks_steps.rb]: @zeljkofilipin yes, main reason for this is that the native WD element does not have another way for this. at least I did not find one.

File: tests/browser/features/sitelinks.feature (hunk 1 above)
Comment (JeroenDeDauw) [inline tests/browser/features/sitelinks.feature]: Err. The frontend uses languages to identify sitelinks?!

Comment (WMDE-Fisch) [inline tests/browser/features/sitelinks.feature]: The combination of language and key of the corresponding wiki is sufficient to identify the sitelink afaik.

File: tests/browser/features/step_definitions/sitelinks_steps.rb (hunk 2 above)
Comment (zeljkofilipin) [inline tests/browser/features/step_definitions/sitelinks_steps.rb]: If watir/page-object api do not trigger all JS events, that is a bug in the apis, and it should be reported. There should not be a need to use the selenium/webdriver api.

Comment (tobijat) [inline tests/browser/features/step_definitions/sitelinks_steps.rb]: What I meant s that the WD element is missing a set method to put text into an input field. That's why we used the send_keys.

Comment (WMDE-Fisch) [inline tests/browser/features/step_definitions/sitelinks_steps.rb]: I avoid all this in #52 by refactoring the usage of @browser.element(css.... to have dynamic elements with the help of indexed properties

**Pull request merged**